### PR TITLE
Magiclysm: Cooking with Poison

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
SUMMARY: Mods "Demon Chitin Cookware and it's uses"

#### Purpose of change
I realized that the mage on the go needed demon spider chitin cookware and what would make sense for this cookware to do?  Remove poisons and help to mix magical ingredients in a large evil looking cauldron.

#### Describe the solution
Adds a recipe book "Cooking with Poison" which includes recipes for the Cauldron, several things that can be made with the cauldron, a new tool quality that hopefully will be helpful when Magiclysm mutation lines become a thing and updates itemgroups to include appropriately. 

#### Describe alternatives you've considered
Using the cauldron to create an army of the undead as a true disciple of the Horned King.

#### Testing
Testing tonight and made changes until it loaded fine and all items could be created without visible errors in game or in Travis.

### CODE REVIEW

**Comment by Maleclypse** [comment]
@KorGgenT @LaVeyanFiend  per our conversation last week, I've created demon chitin cookware.  Let me know what you think works and what doesn't. :)

File: data/mods/Magiclysm/items/tools.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[
+  {
+    "id": "cauldron_demon_chitin",
+    "type": "TOOL",
+    "category": "container",
+    "name": "cauldron of poison",
```
Comment (Qrox) [inline data/mods/Magiclysm/items/tools.json]:
```suggestion
    "name": { "str": "cauldron of poison", "str_pl": "cauldrons of poison" },
```

File: data/mods/Magiclysm/items/tools.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "description": "This cauldron made of demon spider chitin seems to absorb the light.  It will hold 16 liters of material and will absorb poisons from it.  It may have other properties that require discovery.",
+    "weight": "1424 g",
+    "volume": "20 L",
+    "price": 2000,
+    "to_hit": -1,
+    "looks_like": "clay_hydria",
+    "bashing": 1,
+    "material": "demon_chitin",
+    "symbol": ")",
+    "color": "light_gray",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/tools.json]: if it's made of demon chitin, i'd suggest red

Comment (Maleclypse) [reply data/mods/Magiclysm/items/tools.json]: done

File: data/mods/Magiclysm/items/tools.json (hunk 1 above)
Comment (Maleclypse) [reply data/mods/Magiclysm/items/tools.json]: name changed but new name fits naming convention.  Thanks!

File: data/mods/Magiclysm/recipes/cooking.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
+[
+  {
+    "type": "recipe",
+    "result": "purified_meat",
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_MEAT",
+    "skill_used": "spellcraft",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/recipes/cooking.json]: reverse the skill required to cooking being primary and spellcraft being secondary, for both recipes.

File: data/mods/Magiclysm/items/tools.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+[
+  {
+    "id": "cauldron_demon_chitin",
+    "type": "CONTAINER",
+    "category": "other",
+    "name": { "str": "cauldron of purification", "str_pl": "cauldrons of purification" },
+    "description": "This cauldron made of demon spider chitin seems to absorb the light.  It will hold 16 liters of material and will absorb poisons from it.  It may have other properties that require discovery.",
+    "weight": "1424 g",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/tools.json]: sorry i didn't catch this earlier, but for a 20 L object, 3 pounds is very very light.

Comment (Maleclypse) [reply data/mods/Magiclysm/items/tools.json]: I agree, but I used straight math for demon chitin and bowl volume.  If the cauldron contains 16 L of volume that means 4 L of material.  Each chunk of demon chitin is 250 ml so 16 chunks to make the cauldron.  4 L of Demon chitin chunks at 89 grams per 250 ml comes out to 1424 exactly.  I'm good to update it to whatever you think is best but I feel it's important to show my math and why I will need you to let me know what it should actually be?  I can add the bone glue in weight but that only adds 450 g in weight.

**Pull request merged**